packages/core/lib/console-child.js

```diff
@@ -23,41 +24,71 @@ global.crypto = {
 const { network, config, url } = yargs(input[0]).argv;
 const detectedConfig = Config.detect({ network, config });
 
-function getConfiguredNetworkUrl(customConfig) {
+function getConfiguredNetworkUrl(customConfig, isDashboardNetwork) {
+  const defaultPort = isDashboardNetwork
+    ? managedDashboardDefaultPort
+    : managedGanacheDefaultPort;
   const configuredNetworkOptions = {
-    host: customConfig.host || managedGanacheDefaultHost,
-    port: customConfig.port || managedGanacheDefaultPort
+    host: customConfig.host || defaultHost,
+    port: customConfig.port || defaultPort
   };
-  return `http://${configuredNetworkOptions.host}:${configuredNetworkOptions.port}/`;
+  const urlSuffix = isDashboardNetwork ? "/rpc" : "";
+  return `http://${configuredNetworkOptions.host}:${configuredNetworkOptions.port}${urlSuffix}`;
 }
 
 let configuredNetwork;
 
-if (url) {
+const configNetworkWithProvider =
+  detectedConfig.networks[network] && detectedConfig.networks[network].provider;
+
+if (configNetworkWithProvider) {
+  // Use "provider" specified in the network config
+  configuredNetwork = {
+    ...detectedConfig.networks[network],
+    network_id: "*",
+    provider: detectedConfig.networks[network].provider
+  };
+} else if (url) {
   // Use "url" to configure network (implies not "develop")
   configuredNetwork = {
     network_id: "*",
-    url
+    url,
+    provider: function () {
+      return new Web3.providers.HttpProvider(url, {
+        keepAlive: false
+      });
+    }
   };
 } else {
   // Otherwise derive network settings
   const customConfig = detectedConfig.networks[network] || {};
+  const isDashboardNetwork = network === "dashboard" ? true : false;
```

Comment:
```suggestion
  const isDashboardNetwork = network === "dashboard";
```
The result of the comparison is a Boolean